File: solid_principles/python/example1/problemetic_code/employee.py
from typing import List, Any

"""
This class has issues, 
what is the issue:-
1. This class is doing too many things.
2. 
"""

class Employee:
    __id: int
    __name: str

    def __init__(self, id, name):
        self.__id = id
        self.__name = name

    def print_performance_report(self, e: Employee) -> str:
        print('Performance Report of Employee:- {0}'.format(e))

    def compute_salary(self, e: Employee) -> float:
        print('salary of Employee.')

    def update_employee_data(self, e: Employee) -> bool:
        print('Employee Updation.')

    def fetch_employee_data(self, e: Employee) -> List[dict[str, Any]]:
        print('Fetched Employee Data.')



# Driver Code...
e1 = Employee(1, 'dj')
print(e1.id)

File: solid_principles/java/example1/ProblemCode/Employee.java

public class Employee {
    private int id;
    private String name;

    public Employee(int id, String name) {
        this.id = id;
        this.name = name;
    }

    public double getEmployeeSalary() {
        return 1000.0;
    }

    public void printPerformanceReport() {
        System.out.println("Performance Report.");
    }

    public void updateEmployeeData() {
        System.out.println("Updation of Employee Data.");
    }

    public void fetchEmployeeData() {
        System.out.println("Employee Details.");
    }
}

// what is the issue with above class.
// issue1:- This class has many responsibilities.
// It does gettters and setters
// It does printing responsibility.
// It does updation
